examples/playground/src/app.js

```diff
@@ -20,13 +20,17 @@ const INITIAL_TEMPLATE = Object.keys(JSON_TEMPLATES)[0];
 // Set your mapbox token here
 const GOOGLE_MAPS_TOKEN = process.env.GoogleMapsToken; // eslint-disable-line
 
+function isFunctionObject(value) {
+  return typeof value === 'object' && '@@function' in value;
+}
+
 function addUpdateTriggersForAccessors(json) {
   if (!json || !json.layers) return;
 
   for (const layer of json.layers) {
     const updateTriggers = {};
     for (const [key, value] of Object.entries(layer)) {
-      if (key.startsWith('get') && typeof value === 'string') {
+      if ((key.startsWith('get') && typeof value === 'string') || isFunctionObject(value)) {
```

Comment:
I think `@@function` syntax could be supported as generic "hydration" syntax on the top level - not just for accessors - as an equal citizen with `@@type`. 

We could also support `@@class` as an alternative to `@@type` - so you could either hydrate class instances or functions.

(A third option could also just be to have `@@type` support both - look into the JSON configuration and if there is a function registered under that name use that, if there is a class instantiate that. But having separate keyword may make things a little easier for the users).

Reply:
+1, we should do extend this outside of accessors and It was the idea of the original [RFC](https://github.com/visgl/deck.gl/issues/5222)

Reply:
I'm more fan of creating a new @@function and keep the old as `@@type`. Open to have `@class` as an alternative (it's now a better name), but not a priority for me right now.